src/libraries/BitMath.sol

```diff
@@ -3,96 +3,47 @@ pragma solidity ^0.8.20;
 
 /// @title BitMath
 /// @dev This library provides functionality for computing bit properties of an unsigned integer
+/// @author Solady (https://github.com/Vectorized/solady/blob/8200a70e8dc2a77ecb074fc2e99a2a0d36547522/src/utils/LibBit.sol)
 library BitMath {
     /// @notice Returns the index of the most significant bit of the number,
     ///     where the least significant bit is at index 0 and the most significant bit is at index 255
-    /// @dev The function satisfies the property:
-    ///     x >= 2**mostSignificantBit(x) and x < 2**(mostSignificantBit(x)+1)
     /// @param x the value for which to compute the most significant bit, must be greater than 0
     /// @return r the index of the most significant bit
     function mostSignificantBit(uint256 x) internal pure returns (uint8 r) {
```

Comment:
solmates version returns a uint256 - is this valid if we truncate to uint8?

Reply:
the input is a uint256, so the maximum possible "most significant bit" is in index 255 (because we index from 0). The largest number that can be held in a uint8 is 255 - so theres enough bits of response

Reply:
The only difference is that solady returns 256 if no bits are set (which wouldnt fit in a uint8). However we check that in advance by having `require(x>0)`